portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,17 +36,34 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private static final int REQUESTED_COMMENTS_LIMIT = 100;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Sort comments by date from newest to oldest
+    int maxCommentsNumber;
+
+    // Try to convert the value from the 'max-comments' field of the query string to int
+    try {
+      maxCommentsNumber = Integer.parseInt(request.getParameter("max-comments"));
+    } catch (NumberFormatException e) {
+      // Handle the exception
+      maxCommentsNumber = REQUESTED_COMMENTS_LIMIT;
+    }
+
+    // If maxCommentsNumber is negative, fallback to the maximum instead
```

Comment:
If we do in the current way, we are trying to fix problems of clients. This is not a proper behavior. When the input is wrong, we need to throw an exception so that client can get an error and then fix it.

Reply:
Done!

Reply:
Also very nice point, thank you!